Display latest 1.x version next to latest 2.x version

michizhou opened the pull request:
Signed-off-by: Michael Zhou <mizho@alumni.upenn.edu>

### Description
Displays the latest 1.x line version next to the latest 2.x line version on the website homepage
 
### Issues Resolved
#971 

### Check List
- [x] Commits are signed per the DCO using --signoff


By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

krisfreedain commented:
thank you @michizhou - this looks super interesting. 

Here's what I'm seeing in a test of the code
![Screen Shot 2022-10-31 at 11 54 12 AM](https://user-images.githubusercontent.com/4873623/199087416-14171714-3b39-4db5-a5db-befe561b6590.png)

Doesn't seem to show the 1.x line as I assume is intended. 
@nateynateynate @dtaivpp - can you give this a look and provide @michizhou input? thanks

michizhou commented:
@nateynateynate @dtaivpp I wanted to follow up on this, since it currently doesn't show the 1.x version as expected. Can one of you review this and provide input on how this issue can be resolved?

dtaivpp commented on _layouts/homepage.html:
```diff
@@ -70,8 +70,16 @@ <h1>{{ page.secondary.heading }}</h1>
+      {% assign latest = versions | first %}
+      {% for version in versions %}
+          {% assign major_version = version.version | split: "." | first %}
+          {% if major_version == 1 %}
```
Okay so the reason it is not rendering comes down to this. Here you are comparing the string's '1', '2' to the number 1. We could fix this to be '1' but that leaves us with the issue of updating this every time we have a major release which I am not a fan of so I have 2 other recommendations outlined in the above sections.

dtaivpp replied:
3. Finally, we can now compare the current major from the iteration to the previous major. 

```{% if major_version == previous_major %}```

dtaivpp commented on _layouts/homepage.html:
```diff
@@ -70,8 +70,16 @@ <h1>{{ page.secondary.heading }}</h1>
-      {% assign latest = site.versions | sort: 'version_sort' | reverse | first %}
-      <div class="latest-version">{{ page.version_feature.latest_label}} {{ latest.version }} {{ page.version_feature.date_label}} {{ latest.date | date_to_string: "ordinal", "US"}}</div>
+      {% assign versions = site.versions | sort: 'version_sort' | reverse %}
+      {% assign latest = versions | first %}
```
1. First lets add a section below this line. This will calculate what the previous major should be: 

```{% assign previous_major = latest.version | split: "." | first | plus: -1 %}```

Note, the plus here converts the version string to an int which we have to account for later.

dtaivpp commented on _layouts/homepage.html:
```diff
@@ -70,8 +70,16 @@ <h1>{{ page.secondary.heading }}</h1>
+      {% assign versions = site.versions | sort: 'version_sort' | reverse %}
+      {% assign latest = versions | first %}
+      {% for version in versions %}
+          {% assign major_version = version.version | split: "." | first %}
```
2. Now we need to ensure the major_version is an integer. We can do that by putting a pipe and plus zero right after first.

```{% assign major_version = version.version | split: "." | first | plus: 0 %}```

dtaivpp reviewed (changes requested):
Thanks for your patience on this. It took a lot longer than I had expected. I added the changes needed to get this PR accepted. Thank you for all your work on this! I know a log of people have been asking for it.

michizhou commented:
@dtaivpp Thank you for taking a look at this and giving your recommended changes. I have incorporated them into the PR and updated it accordingly. Please let me know if any additional changes are needed.

michizhou commented:
@dtaivpp Thank you for taking a look at this and giving your recommended changes! I have incorporated them into the PR and updated it accordingly. Please let me know if any additional changes are needed.

dtaivpp reviewed (approved):
Pulled local and tested and looks good to me!
